tests/components/upnp/test_init.py

```diff
@@ -200,3 +201,37 @@ async def test_async_setup_entry_force_poll(
     assert await hass.config_entries.async_setup(entry.entry_id) is True
 
     mock_igd_device.async_subscribe_services.assert_not_called()
+
+
+@pytest.mark.usefixtures(
+    "ssdp_instant_discovery",
+    "mock_get_source_ip",
+    "mock_mac_address_from_host",
+)
+async def test_async_setup_entry_force_poll_subscribe_error(
+    hass: HomeAssistant, mock_igd_device: IgdDevice
+) -> None:
+    """Test async_setup_entry where subscribing fails."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=TEST_USN,
+        data={
+            CONFIG_ENTRY_ST: TEST_ST,
+            CONFIG_ENTRY_UDN: TEST_UDN,
+            CONFIG_ENTRY_ORIGINAL_UDN: TEST_UDN,
+            CONFIG_ENTRY_LOCATION: TEST_LOCATION,
+            CONFIG_ENTRY_MAC_ADDRESS: TEST_MAC_ADDRESS,
+        },
+        options={
+            CONFIG_ENTRY_FORCE_POLL: False,
+        },
+    )
+
+    # Subscribing partially succeeds, but not completely.
+    # Unsubscribing will fail for the subscribed services afterwards.
+    mock_igd_device.async_subscribe_services.side_effect = UpnpCommunicationError
+    mock_igd_device.async_unsubscribe_services.side_effect = UpnpCommunicationError
+
+    # Load config_entry, should still be able to load, falling back to polling/the old functionality.
+    entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(entry.entry_id) is True
```

Comment:
Should we test if we actually poll? Or is this already tested elsewhere?

Reply:
The issue itself is about the initialisation of the component. But it should be good to test this, yes. I'll check later if this is tested elsewhere.

Reply:
Fixed.